### PR TITLE
LPS-146237

### PR DESCRIPTION
**Pull request by javalosjr96**
Ticket:
https://issues.liferay.com/browse/LPS-146237

Original PR:
https://github.com/liferay-lima/liferay-portal/pull/2685
Follow-Up PR:
https://github.com/liferay-lima/liferay-portal/pull/2696

Original Notes:

> Issue:
> 
> PDF's cannot be view in the recycle bin.
> 
> Cause:
> When a PDF is previewed normally in Documents and Media, the source URL(baseImageURL)
> (e.g. http://localhost:8080/documents/20121/0/Test.pdf/a25f2a25-ae2d-aaa9-ee47-4208f5b07001?version=1.0&t=1643171833892&previewFileIndex=)
> has an HTTP parameter for the page index that is appended when the preview is viewed in DocumentPreview.es.js.
> 
> 					 						className={`preview-file-document ${
> 							!expanded && 'preview-file-document-fit'
> 						}`}
> 						src={`${baseImageURL}${currentPage}`}
> 					/>
> When the PDF is moved to the recycle bin, the baseImageURL changes. (e.g.http://localhost:8080/documents/20121/0/Test.pdf/a25f2a25-ae2d-aaa9-ee47-4208f5b07001?version=1.0&t=1643171833892&previewFileIndex=&status=8) and a status parameter is added.
> However, the logic remains the same and the page index is appended incorrectly to the end of the baseImageURL causing the 404 error.
> 
> Solution:
> Created a method that inserts the page index in the correct position.
> 
> There was also promise errors that were thrown due to the method call loadPage having similar logic which is also resolved.

### CODE REVIEW

**Comment by javalosjr96** [comment]
ci:tes:sf

**Comment by javalosjr96** [comment]
ci:test:relevant

**Comment by javalosjr96** [comment]
ci:test:sf

**Comment by javalosjr96** [comment]
@boton

File: modules/apps/document-library/document-library-preview-document/src/main/resources/META-INF/resources/preview/js/DocumentPreviewer.es.js [hunk 1]
```diff
@@ -95,11 +95,19 @@ const DocumentPreviewer = ({baseImageURL, initialPage, totalPages}) => {
...
+
+		return imageURL.toString();
+	};
+
 	const loadPage = (page) => {
 		let pagePromise = loadedPages[page] && loadedPages[page].pagePromise;
 
 		if (!pagePromise) {
-			pagePromise = imagePromise(`${baseImageURL}${page}`).then(() => {
+			pagePromise = imagePromise(`${createImageURL(page)}`).then(() => {
```
Comment (boton) [inline modules/apps/document-library/document-library-preview-document/src/main/resources/META-INF/resources/preview/js/DocumentPreviewer.es.js:110]: Please remove and use directly the function `imagePromise(createImageURL(page))`

File: modules/apps/document-library/document-library-preview-document/src/main/resources/META-INF/resources/preview/js/DocumentPreviewer.es.js [hunk 2]
```diff
@@ -214,7 +222,7 @@ const DocumentPreviewer = ({baseImageURL, initialPage, totalPages}) => {
 						className={`preview-file-document ${
 							!expanded && 'preview-file-document-fit'
 						}`}
-						src={`${baseImageURL}${currentPage}`}
+						src={`${createImageURL(currentPage)}`}
```
Comment (boton) [inline modules/apps/document-library/document-library-preview-document/src/main/resources/META-INF/resources/preview/js/DocumentPreviewer.es.js:225]: Same here:

`src={createImageURL(currentPage)}`

**Review by boton — CHANGES_REQUESTED** [review]

**Comment by javalosjr96** [comment]
@boton Changes Made.

**Comment by boton** [comment]
ci:forward

**Pull request closed without merging**